### DISCUSSION TITLE
how can i call another dispatch when first dispatch done?

### DISCUSSION THREAD

**Discussion by giovannidias1** [ngrx/platform#3535] [Q&A] [answered]
I have a first diaspatch that send a question and second that load a questions list 

```
    this.store.dispatch(sendAnswer({ answer: data }));
    this.store.dispatch(loadQuestion({ slug: this.slug, take: 5 }));
```
but the second one is called when the first is not finished yet

what's the best way to do that?

**Comment by ramarivera** [comment, marked as answer]
Assuming that "sending" a question ends up being an API call (or some "long" running process),  you could have an effect listening to `sendAnswer`, and when it finishes processing it (whatever that means in your context), then this effect would dispatch another action e.g. `answerSent`.

You could then do something like

```
this.store.dispatch(sendAnswer({ answer: data }));

this.store.select(someSelectorThatReadsThatDataWasUpdatedBasedOnYourAnswer(data)).
               .pipe(take(1))
               .subscribe(() => { 
                    this.store.dispatch(loadQuestion({ slug: this.slug, take: 5 }));
               });
```

Not claiming it's the best way, just a way
